client/src/js/components/App.js

```diff
@@ -64,6 +65,7 @@ const App = ({ isFirstTimeAccess }) => {
               <Route exact path="/syllabus" component={Syllabus} />
               <Route exact path="/roomfinder" component={RoomFinder} />
               <Route exact path="/bus" component={Bus} />
+              <Route exact path="/comments/:courseKey" component={Comments} />
```

Comment:
Url path design is a very interesting topic. Please read this [3 min article](https://medium.com/@fullsour/when-should-you-use-path-variable-and-query-parameter-a346790e8a6d). It's helpful for our discussion. 👍 

Currently, we are getting evaluations by their course keys. But in the future, we want to identify an evaluation by its **id** (when a user wants to view, update, create an evaluation). This means `/courseEvals/:xxx` should be reserved for id.

I recommend `path="/courseEvals" component={CourseEvals}`. Inside `CourseEvals`, you have to parse the query string. The query parameter should be `?courseId=xxxxx` because **each course will have a unique eval page** unless it doesn't have any evaluations. Imagine we use `courseKey`, what will happen when a user visits `courseEvals?courseKey=xxx` directly? Our system will not be able to identify which course the user was looking at. This is why we need `courseId`.

So the logic goes like this:
1. A user clicks the evaluation button or visits `courseEvals?courseId=xxx` directly (maybe a friend shared this url with him).
2. By parsing the query parameter, we get the `courseId`, which is `xxx`. We know which course this id points to, so we can show the course on top of the page. 
4. We also know the course code (e.g. `yyy`) and course key (e.g. `zzz`) for this course. We then call the backend and get all evaluations with course code equals to `yyy`. 
5. Among all the evaluations we got, for the ones that have the same course key (which is `zzz`), we display them in the "Evaluations" section. 
6. The evaluations that do not have the same course key are from related courses. We display them in the "Evaluations for similar courses" section. 

I don't think we can display similar courses because it requires us to know their ids. Let's discuss this in the future.

Reply:
- I have implemented the query parameter to pass course ID. The above 2 tasks are also done. Please check!
- If we can retrieve related courses data with **course code**, I think it won't be a problem to display related courses with their evaluations in the right column.
- I have defined functions to retrieve the data from DB after Backend is prepared. For now these functions just deal with the dummy data I directly wrote inside the code.